### PR TITLE
Implement GLib::Regex#replace

### PR DESCRIPTION
**Pull request by cedlemo**

### CODE REVIEW

File: glib2/ext/glib2/rbglib_regex.c [hunk 1]
```diff
@@ -260,6 +260,47 @@ rg_match_all(gint argc, VALUE *argv, VALUE self)
...
+                                      string,
+                                      string_len,
+                                      start_position,
+                                      replacement,
+                                      match_options,
+                                      &error);
+    if (error)
+        RAISE_GERROR(error);
+
+    return CSTR2RVAL(modified_string);
```
Comment (kou) [inline glib2/ext/glib2/rbglib_regex.c]: Use `CSTR2RVAL_FREE()`. The current code leaks a memory.

File: glib2/test/test-regex.rb [hunk 2]
```diff
@@ -250,4 +250,16 @@ def test_regex_escape_string
       assert_equal(["toto ", ""], splited_strings)
     end
   end
+
+  sub_test_case "replace" do
+    test "simple" do
+      regex = GLib::Regex.new("\s")
```
Comment (kou) [inline glib2/test/test-regex.rb]: It should be `"\\s"`.

**Comment by cedlemo** [comment]
Modifications done.

**Comment by kou** [comment]
Thanks!

**Pull request merged**